crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cheated_syscalls.rs

```diff
@@ -240,6 +243,76 @@ pub fn get_block_hash_syscall(
     Ok(GetBlockHashResponse { block_hash })
 }
 
+#[allow(clippy::needless_pass_by_value)]
+pub fn storage_read(
+    request: StorageReadRequest,
+    _vm: &mut VirtualMachine,
+    syscall_handler: &mut SyscallHintProcessor<'_>,
+    cheatnet_state: &mut CheatnetState,
+    _remaining_gas: &mut u64,
+) -> SyscallResult<StorageReadResponse> {
+    let original_storage_address = syscall_handler.base.call.storage_address;
+    maybe_modify_storage_address(syscall_handler, cheatnet_state);
+
+    let value = syscall_handler
+        .base
+        .storage_read(request.address)
+        .inspect_err(|_| {
+            // Restore state on error before bubbling up
+            syscall_handler.base.call.storage_address = original_storage_address;
```

Comment:
This maybe could be wrapped in the function just so it's clearer what it does without needing a comment.

And we could reuse it below as well.

Reply:
@ddoktorski up to you then